spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -2079,11 +2073,10 @@ private void batchAfterRollback(final ConsumerRecords<K, V> records,
 		private RuntimeException doInvokeBatchListener(final ConsumerRecords<K, V> records, // NOSONAR
 				List<ConsumerRecord<K, V>> recordList) {
 
-			Object sample = startMicrometerSample();
-			try {
+			Observation observation = startObserving();
+			try (Scope scope = observation.openScope()) {
```

Comment:
Thanks for the doc, Gary!

So, this one was my question: looks like `lowCardinality` is about metrics, the `highCardinality` for tracing.
And according this migration docs they say: https://github.com/micrometer-metrics/micrometer/wiki/Migrating-to-new-2.0.x-Observation-API#what-is-this-observationscope-about

> This might not be relevant from only Metrics point of view, however it’s critical from Tracing / Logging perspective.

Therefore my concern if you are going to stay just with metrics or looking forward for tracing as well.

Reply:
As I said on Slack yesterday...

>Next up: tracing.